craft_application/application.py

```diff
@@ -211,6 +211,7 @@ def _configure_services(
         self,
         platform: str | None,  # noqa: ARG002 (Unused method argument)
         build_for: str | None,  # noqa: ARG002 (Unused method argument)
+        provider_name: str | None,
```

Comment:
This is a breaking change - let's get an exclamation mark in that commit description :-)